includes/Admin/Admin_AJAX.php

```diff
@@ -278,6 +239,48 @@ public function run_checks() {
 		);
 	}
 
+
+	/**
+	 * Get the AJAX_Runner runner.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return AJAX_Runner
+	 */
+	protected function get_runner() {
+		$runner = Plugin_Request_Utility::get_runner();
+
+		if ( ! ( $runner instanceof AJAX_Runner ) ) {
+			$runner = new AJAX_Runner();
+		}
```

Comment:
This is a functional change we shouldn't make: This method should only instantiate a new `AJAX_Runner` if `Plugin_Request_Utility::get_runner()` returns `null`. Otherwise, if the runner returned from that method itself is not an `AJAX_Runner`, we should throw an exception and then the underlying AJAX callback should terminate with a JSON error, similar to how that works today.

Reply:
Why would we want to do that? If it returns null, it is okay to create a new runner right? If `get_runner` returns a CLI_Runner, then error? Why error here? What is the benefit of erroring when we can just create an instance?